drivers/mma7660/mma7660.c

```diff
@@ -0,0 +1,246 @@
+    }
+
+    if (mma7660_config_interrupts(dev, MMA7660_INTSOURCE_NONE) != MMA7660_OK) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_set_mode(const mma7660_t *dev, uint8_t active,
+                     uint8_t autowake, uint8_t autosleep, uint8_t prescale)
+{
+    char reg;
+    int rv;
+    reg = (active << MODE_ACTIVE_SHIFT)       |
+          (autowake << MODE_AUTOWAKE_SHIFT)   |
+          (autosleep << MODE_AUTOSLEEP_SHIFT) |
+          (prescale << MODE_PRESCALE_SHIFT)   |
+          MODE_INTERRUPT_DEFAULT;
+    i2c_acquire(DEV_I2C);
+    rv = i2c_write_reg(DEV_I2C, DEV_ADDR, MMA7660_MODE, reg);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_read_tilt(const mma7660_t *dev, uint8_t *res)
+{
+    int rv;
+    i2c_acquire(DEV_I2C);
+    rv = i2c_read_reg(DEV_I2C, DEV_ADDR, MMA7660_TILT, (char*)res);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_READ_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_write_sleep_count(const mma7660_t *dev, uint8_t sleep)
+{
+    int rv;
+    i2c_acquire(DEV_I2C);
+    rv = i2c_write_reg(DEV_I2C, DEV_ADDR, MMA7660_SPCNT, sleep);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_config_interrupts(const mma7660_t *dev, uint8_t isource_flags)
+{
+    int rv;
+    i2c_acquire(DEV_I2C);
+    rv = i2c_write_reg(DEV_I2C, DEV_ADDR, MMA7660_INTSU, isource_flags);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_config_samplerate(const mma7660_t *dev, uint8_t amsr, uint8_t awsr, uint8_t filt)
+{
+    int rv;
+    char ch = amsr | awsr | (filt << SR_FILT_SHIFT);
+    i2c_acquire(DEV_I2C);
+    rv = i2c_write_reg(DEV_I2C, DEV_ADDR, MMA7660_SR, ch);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_config_pdet(const mma7660_t *dev, uint8_t pdth, uint8_t enabled_axes) {
+    int rv;
+    char ch = pdth | ((~enabled_axes) & PDET_AXIS_MASK);
+    i2c_acquire(DEV_I2C);
+    rv = i2c_write_reg(DEV_I2C, DEV_ADDR, MMA7660_PDET, ch);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_config_pd(const mma7660_t *dev, uint8_t pd) {
+    int rv;
+    i2c_acquire(DEV_I2C);
+    rv = i2c_write_reg(DEV_I2C, DEV_ADDR, MMA7660_PD, (char)pd);
+    i2c_release(DEV_I2C);
+    if (rv != 1) {
+        return -MMA7660_I2C_WRITE_ERR;
+    }
+
+    return MMA7660_OK;
+}
+
+int mma7660_read_counts(const mma7660_t *dev, int8_t *x, int8_t *y, int8_t *z)
+{
+    int retries = 6;
+    char t;
+    i2c_acquire(DEV_I2C);
+
+    while (retries > 0) {
+        if (i2c_read_reg(DEV_I2C, DEV_ADDR, MMA7660_XOUT, &t) != 1) {
```

Comment:
from the data sheet I see that x|y|zout are placed in row, so IMHO it would make sense to have 

```
uint8_t tmp[3];
...
 if (i2c_read_regs(DEV_I2C, DEV_ADDR, MMA7660_XOUT, &tmp[0], 3) != 3) {
...
```

or not,

Reply:
I didn't make this change yet because I vaguely remember having a reason for doing it separately and I don't have the time to do in depth testing again. Is it important?

Reply:
at least its not blocking from my side, but for sake of simplicity and efficiency we (you? 😉) should look into this later on.

Reply:
Agreed